app/payment-link-v2/reference/post-index.controller.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+const lodash = require('lodash')
+
+const { response } = require('../../utils/response')
+const { getSessionVariable } = require('../../utils/cookie')
+const paths = require('../../paths')
+const { validateReference } = require('../../utils/validation/form-validations')
+const replaceParamsInPath = require('../../utils/replace-params-in-path')
+const { setSessionVariable } = require('../../utils/cookie')
+const getBackLinkUrl = require('./get-back-link-url')
+const isAPotentialPan = require('./is-a-potential-pan')
+
+const PAYMENT_REFERENCE = 'payment-reference'
+
+function validateReferenceFormValue (reference, res) {
+  const errors = {}
+  const referenceValidationResult = validateReference(reference)
+  if (!referenceValidationResult.valid) {
+    errors[PAYMENT_REFERENCE] = res.locals.__p(referenceValidationResult.messageKey)
+  }
+
+  return errors
+}
+
+function getNextPageUrl (productPrice, sessionAmount, reference) {
+  if (productPrice || sessionAmount) {
+    return paths.paymentLinksV2.confirm
+  } else if (isAPotentialPan(reference)) {
```

Comment:
This should be the first case, as we always want to ask the user to check their reference number if it is potentially a PAN

Reply:
Good spot - updated this and added a unit test.